service/worker/deployment/deployment_client.go

```diff
@@ -369,51 +338,396 @@ func (d *DeploymentClientImpl) ListDeployments(ctx context.Context, namespaceEnt
 
 }
 
-// GenerateStartWorkflowPayload generates start workflow execution payload
-func (d *DeploymentClientImpl) generateStartWorkflowPayload(namespaceEntry *namespace.Namespace, deployment *deploymentpb.Deployment) (*commonpb.Payloads, error) {
-	workflowArgs := &deploymentspb.DeploymentWorkflowArgs{
+func (d *DeploymentClientImpl) SetCurrentDeployment(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	deployment *deploymentpb.Deployment,
+	updateMetadata *deploymentpb.UpdateDeploymentMetadata,
+	identity string,
+	requestID string,
+) (*deploymentpb.DeploymentInfo, *deploymentpb.DeploymentInfo, error) {
+	updatePayload, err := sdk.PreferProtoDataConverter.ToPayloads(&deploymentspb.SetCurrentDeploymentArgs{
+		Identity:       identity,
+		BuildId:        deployment.BuildId,
+		UpdateMetadata: updateMetadata,
+		RequestId:      requestID,
+	})
+	outcome, err := d.updateWithStartDeploymentSeries(
```

Comment:
is the "start" part to allow the deployment to be set as default even if the series and deployment do not exist yet?

Reply:
Yes, it seems easiest to just use update-with-start everywhere, so everything just gets created on demand. Do you think the semantics should be different?

On a similar note, a new deployment series should have no current, right? (Not, e.g., the first deployment created in it?)

Reply:
I think the semantic makes sense, we may want to put it behind a `--force` flag in the future because user should run their workers first.

>  a new deployment series should have no current

yes.